segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScaleProcessor.java

```diff
@@ -111,7 +112,15 @@ private void bootstrapRequestWriters() {
                 })
                 .runAsync(() -> {
                     if (clientFactory.get() == null) {
-                        clientFactory.compareAndSet(null, ClientFactory.withScope(NameUtils.INTERNAL_SCOPE_NAME, configuration.getControllerUri()));
+                        ClientFactory factory = null;
+                        if (configuration.isAuthEnabled()) {
+                            factory = ClientFactory.withScope(NameUtils.INTERNAL_SCOPE_NAME, configuration.getControllerUri(),
+                                    new PravegaDefaultCredentials(configuration.getAuthPasswd(), configuration.getAuthUsername()),
```

Comment:
I am concerned about this because it explicitly uses `PravegaDefaultCredentials` which is a specialization of `PravegaCredentials`.   I believe I understand that the goal here is for the segment store to talk to the controller on its own behalf.  A sort of "service account" basically.

On the controller side, depending on which auth providers are configured, the `PravegaDefaultAuthHandler` may not be there (am I wrong in that assumption?) to handle these specific kinds of credentials.

Or, is the idea that the `PravegaDefaultAuthHandler` on the controller side, is ALWAYS there; and other auth providers would be configured in _addition_ to it?

Related to the above, that would mean that the password file for example, is always required, and required to contain at the very least that one service user.

For some reason, I seem to remember from the PDP that authentication between pravega components would be done using certs instead of this?   Let me know if I misunderstood.

Reply:
It is necessary that `PravegaDefaultAuthHandler` will always be there (as it is bundled with `controller`).
I thought about making the creds used by `segmentstore` to access controller pluggable too. But I think it is not necessary. We will need to build another pluggable mechanism on the `segmentstore` for the creds. Also we know the kind of access this service account needs (access to `_system` scope). I do not see any need to allow custom implementations to override that. Considering the complexity and the value provided, I decided to go ahead with only using `PravegaDefaultCredentials` explicitly.

Reply:
Understood.  I agree since this is internal to Pravega, the value to make this pluggable is low.  I think if we have a README or deployment guide that clearly illustrates that one needs to create a password file (and how), with a service user, and the special acl permissions on the _system scope, regardless of any other auth provider that may be involved, I think we are ok for now.

One thing though: this just puts a little more spotlight on the PravegaDefaultAuthHandler since it's now not just the hello world default implementation of what implementors may choose as a start/template, and override; but it's actually THE auth handler implementation used for the internal communication of pravega (segment store ---> controller at least) no matter what.

It means, if I understood correctly, that even if I add a super secure implementation of `PravegaAuthHandler`, with its own repository of users etc., one time temporary passwords etc.,  at the end of the day, the username/password of that one super powerful service user will still be there and used, by way of the `PravegaDefaultAuthHandler`, so it surely needs to be initialized in production to some very secure password, with future enhancements accounting for updating/rotating that password; perhaps changing the method to some HMAC signing etc.    So the overall security protection for the controller is as good as the `PravegaDefaultAuthHandler` can be regardless of additional providers.

(just thinking ahead and making sure I'm correctly assessing the current design/implementation).

Again I think the current approach is fine for this release and NOT unlike many other systems that also use powerful local accounts for bootstrapping.